### PR TITLE
[FIX] web_widget_char_switchcase: keep caret position

### PR DESCRIPTION
**Pull request by SimoRubi**
Keep caret position after changing the value

### CODE REVIEW

**Review by eLBati — APPROVED** [review]
tested

**Review by astirpe — COMMENTED** [review]
In Travis I read this:

```
ERROR openerp_test odoo.tests.common: phantomjs: error
```

**Comment by SimoRubi** [comment]
Thanks @astirpe I still had to check Travis, now it should be good

File: web_widget_char_switchcase/static/src/js/web_widget_char_switchcase.js [hunk 1]
```diff
@@ -21,9 +21,18 @@ odoo.define('web_widget_char_switchcase', function (require) {
                 console.error(this.current_transformation + ' case unknown');
         },
         _onKeyUp: function (event) {
+            // save caret position
```
Comment (astirpe) [inline web_widget_char_switchcase/static/src/js/web_widget_char_switchcase.js]: Please check travis warnings:
```
web_widget_char_switchcase/static/src/js/web_widget_char_switchcase.js:24: [W7903(javascript-lint), ] Comments should not begin with a lowercase character [Error/capitalized-comments]
```

File: web_widget_char_switchcase/static/src/js/web_widget_char_switchcase.js [hunk 2]
```diff
@@ -21,9 +21,18 @@ odoo.define('web_widget_char_switchcase', function (require) {
...
+            var input = this.$input[0];
+            var start = input.selectionStart;
+            var end = input.selectionEnd;
+
```
Comment (astirpe) [inline web_widget_char_switchcase/static/src/js/web_widget_char_switchcase.js]:
```
web_widget_char_switchcase/static/src/js/web_widget_char_switchcase.js:28: [W7903(javascript-lint), ] Trailing spaces not allowed. [Error/no-trailing-spaces]
```

**Review by astirpe — CHANGES_REQUESTED** [review]

**Comment by SimoRubi** [comment]
@astirpe now travis is :heavy_check_mark:

**Comment by astirpe** [comment]
I mean here: https://travis-ci.org/OCA/web/jobs/443104196

**Comment by SimoRubi** [comment]
@astirpe I have fixed most JSLint warnings, Travis now is :heavy_check_mark: (as it was before, since JSLint doesn't make the build fail)

**Review by astirpe — APPROVED** [review]
Thank you!

**Comment by SimoRubi** [comment]
Fixed up

**Comment by SimoRubi** [comment]
@simahawk sorry to bother you again, but you already know this module so well!
Could you please review? I also solved most of the JSLint warnings!
Thanks

**Review by simahawk — APPROVED** [review]

**Comment by simahawk** [comment]
thanks for keeping commits clear ;)

**Pull request merged**